File: BOJ/CPP/단계별로_풀어보기/04_while문/BOJ_10952.cpp
/*
	BOJ 10952 : A+B - 5

	author : kdgyun
	link https://st-lab.tistory.com
	link https://github.com/kdgyun
	Created by kdgyun
	Copyright © 2021 kdgyun. All rights reserved.
*/

#include <iostream>

using namespace std;

int main(int argc, char const *argv[]) {

	ios_base::sync_with_stdio(false);
	cin.tie(NULL);
	int A, B;

	while (true) {
		cin >> A >> B;
		if (A == 0 && B == 0) { // A와 B가 모두 0이라면 while문 종료
			break;
		}
		cout << A + B << "\n";
	}

	return 0;
}

File: BOJ/Java/단계별로_풀어보기/B_03_반복문/BOJ_10952.java
package 단계별로_풀어보기.B_03_반복문;

/**
 * BOJ 10952 : A+B - 5
 * 
 * @author kdgyun
 * {@link https://st-lab.tistory.com}
 * {@link https://github.com/kdgyun}
 *
 */

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;

public class BOJ_10952 {

    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        StringBuilder sb = new StringBuilder();

        while (true) {

            String str = br.readLine();

            int A = str.charAt(0) - 48;
            int B = str.charAt(2) - 48;

            if (A == 0 && B == 0) {
                break;
            }
            sb.append((A + B)).append('\n');
        }
        System.out.print(sb);
    }
}
